### PR TITLE
docs: replace types in `Validation\Rule`, `Validation\Validation` and `Validation\ValidationInterface`

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
See #6310 

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Validation/ValidationInterface.php [hunk 1]
```diff
@@ -31,7 +31,7 @@ public function run(?array $data = null, ?string $group = null): bool;
      * Check; runs the validation process, returning true or false
      * determining whether or not validation was successful.
      *
-     * @param mixed    $value  Value to validation.
+     * @param object|array|string|int|float|bool|null $value  Value to validation.
```
Comment (paulbalandan) [inline system/Validation/ValidationInterface.php:34]:
```suggestion
     * @param object|array|string|int|float|bool|null $value  Value to validate.
```

File: system/Validation/ValidationInterface.php [hunk 2]
```diff
@@ -31,7 +31,7 @@ public function run(?array $data = null, ?string $group = null): bool;
...
      * @param string   $rule   Rule.
      * @param string[] $errors Errors.
```
Comment (paulbalandan) [inline system/Validation/ValidationInterface.php:36]: Useless comments.
```suggestion
     * @param string[] $errors
```

File: system/Validation/Rules.php [hunk 3]
```diff
@@ -209,7 +209,7 @@ public function not_in_list(?string $value, string $list): bool
     }
 
     /**
-     * @param mixed $str
+     * @param object|array|string|int|float|null $str
```
Comment (kenjis) [inline system/Validation/Rules.php:212]: What about `bool`?

File: system/Validation/ValidationInterface.php (hunk 1 above)
Comment (ping-yee) [reply system/Validation/ValidationInterface.php]: Done.

File: system/Validation/ValidationInterface.php (hunk 2 above)
Comment (ping-yee) [reply system/Validation/ValidationInterface.php]: I have been changed.
Appreciated that!

File: system/Validation/Rules.php (hunk 3 above)
Comment (ping-yee) [reply system/Validation/Rules.php]: Oh, I miss this.
I have been maked up.
Appreciated that!

**Review by paulbalandan — APPROVED** [review]

**Review by kenjis — APPROVED** [review]

**Pull request merged**